parser-typechecker/src/Unison/Codebase/Type.hs

```diff
@@ -174,7 +175,13 @@ data Codebase m v a = Codebase
     namesAtPath :: Path -> m ScopedNames,
     -- Updates the root namespace names index.
     -- This isn't run automatically because it can be a bit slow.
-    updateNameLookup :: m ()
+    updateNameLookup :: m (),
+    -- | The SQLite connection this codebase closes over.
+    --
+    -- At one time the codebase was meant to abstract over the storage layer, but it has been cumbersome. Now we prefer
+    -- to interact with SQLite directly, and so provide this temporary escape hatch, until we can eliminate this
+    -- interface entirely.
+    connection :: Sqlite.Connection
```

Comment:
Is it feasible to provide `withConnection :: (Connection -> m r) -> m r` instead?
Exposing the connection directly allows the connection to leak out of the lifespan of the codebase itself, at which point the connection will have been closed.